ssl-config/src/test/java/com/palantir/ssl/SslSocketFactoriesConnectionTests.java

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.ssl;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.palantir.remoting.ssl.KeyStoreConfiguration;
+import com.palantir.remoting.ssl.SslConfiguration;
+import com.palantir.remoting.ssl.SslSocketFactories;
+import com.palantir.remoting.ssl.TrustStoreConfiguration;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSocketFactory;
+import jersey.repackaged.com.google.common.base.Throwables;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+/**
+ * Tests for {@link SslSocketFactories} that test that SSL connections
+ * can be established properly.
+ */
+public final class SslSocketFactoriesConnectionTests {
+
+    private enum ClientAuth {
+        WITH_CLIENT_AUTH,
+        NO_CLIENT_AUTH,
+    }
+
+    @Rule
+    public Timeout testTimeout = Timeout.seconds(5);
+
+    @Test
+    public void testSslNoClientAuthenticationJks() {
+        TrustStoreConfiguration serverTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration serverKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.SERVER_KEY_STORE_JKS_PATH)
+                .password(TestConstants.SERVER_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trust(serverTrustStoreConfig)
+                .key(serverKeyStoreConfig)
+                .build();
+
+        TrustStoreConfiguration clientTrustStoreConfig = TrustStoreConfiguration.builder()
```

Comment:
I suspect that the 90% use-case is to just specify a trust store path, no password, and default to JKS. this would call for TrustStoreConfiguration.of(Path path) + builders for the remaining 10%.